src/shared/unit-file.c

```diff
@@ -349,10 +357,27 @@ int unit_file_build_name_map(
                                 log_debug("%s: normal unit file: %s", __func__, dst);
                         }
 
-                        r = hashmap_put_strdup(&ids, de->d_name, dst);
-                        if (r < 0)
-                                return log_warning_errno(r, "Failed to add entry to hashmap (%s→%s): %m",
-                                                         de->d_name, dst);
+                        if (revisit_masked) {
+                                if (!path_equal(dst, "/dev/null")) {
+                                        r = hashmap_put_strdup(&withdrawals, de->d_name, dst);
```

Comment:
hmm, am i blind? where is masked_ids itself ever allocated?

Reply:
```
masked_ids = set_new(&string_hash_ops);
if (!masked_ids)
    return log_oom();
```